profile/src/main/java/com/scottlogic/deg/profile/reader/ValidatingProfileReader.java

```diff
@@ -61,9 +63,12 @@ public Profile read() throws IOException {
     }
 
     private void validateFieldsAreTyped(ProfileFields fields) {
-        Optional<Field> untyped = fields.stream().filter(field -> field.type == null).findFirst();
-        if (untyped.isPresent()){
-            throw new InvalidProfileException("Field [" + untyped.get().name + "]: is not typed; add its type to the field definition");
+        List<Field> untyped = fields.stream().filter(field -> field.type == null).collect(Collectors.toList());
+        if (untyped.size() == 1) {
+            throw new InvalidProfileException("Field [" + untyped.iterator().next().name + "]: is not typed; add its type to the field definition");
+        }
+        if (!untyped.isEmpty()) {
+            throw new InvalidProfileException("Fields " + untyped.stream().map(f->f.name).reduce((s, z) -> s + ", " + z).get() + " are not typed; add their type to the field definition");
```

Comment:
This line is really long - could we break it into two lines? This could be done by either using a variable, or by using `String.format` with new lines.